src/main/java/at/ac/tuwien/damap/rest/persons/orcid/ORCIDMapper.java

```diff
@@ -0,0 +1,136 @@
+package at.ac.tuwien.damap.rest.persons.orcid;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import at.ac.tuwien.damap.enums.EIdentifierType;
+import at.ac.tuwien.damap.rest.dmp.domain.ContributorDO;
+import at.ac.tuwien.damap.rest.dmp.domain.IdentifierDO;
+import at.ac.tuwien.damap.rest.persons.orcid.models.ORCIDExpandedSearchPerson;
+import at.ac.tuwien.damap.rest.persons.orcid.models.ORCIDExpandedSearchResult;
+import at.ac.tuwien.damap.rest.persons.orcid.models.ORCIDRecord;
+import at.ac.tuwien.damap.rest.persons.orcid.models.base.ORCIDAffiliation;
+import at.ac.tuwien.damap.rest.persons.orcid.models.base.ORCIDDate;
+import at.ac.tuwien.damap.rest.persons.orcid.models.base.ORCIDEmail;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class ORCIDMapper {
+    public ContributorDO mapExpandedSearchPersonEntityToDO(ORCIDExpandedSearchPerson orcidPerson,
+            ContributorDO contributorDO) {
+
+        contributorDO.setId(null);
+        contributorDO.setFirstName(orcidPerson.getGivenNames());
+        contributorDO.setLastName(orcidPerson.getFamilyNames());
+
+        String firstMail = orcidPerson.getEmails().isEmpty() ? null : orcidPerson.getEmails().get(0);
+        contributorDO.setMbox(firstMail);
+
+        String firstAffiliation = orcidPerson.getAffiliations().isEmpty() ? null
+                : orcidPerson.getAffiliations().get(0);
+        contributorDO.setAffiliation(firstAffiliation);
+
+        IdentifierDO identifierContributorDO = new IdentifierDO();
+        identifierContributorDO.setIdentifier(orcidPerson.getOrcidId());
+        identifierContributorDO.setType(EIdentifierType.ORCID);
+
+        contributorDO.setPersonId(identifierContributorDO);
+
+        return contributorDO;
+    }
+
+    public ContributorDO mapRecordEntityToPersonDO(ORCIDRecord orcidRecord,
+            ContributorDO contributorDO) {
+
+        contributorDO.setId(null);
+        contributorDO.setFirstName(orcidRecord.getPerson().getName().getGivenNames().getValue());
+        contributorDO.setLastName(orcidRecord.getPerson().getName().getFamilyName().getValue());
+
+        var primaryMail = orcidRecord.getPerson().getEmails().getEmail().stream().filter(ORCIDEmail::isPrimary)
+                .findFirst();
+        contributorDO.setMbox(primaryMail.isPresent() ? primaryMail.get().getEmail() : null);
```

Comment:
If a user only has a secondary email (e.g. the main one is private), then the second email shows up in the frontend. 
<img width="984" alt="Before saving" src="https://github.com/tuwien-csd/damap-backend/assets/80054001/1da3ca13-cae8-4653-8631-b31fee3b7703">
But after saving the dmp, it is gone.
<img width="976" alt="After saving" src="https://github.com/tuwien-csd/damap-backend/assets/80054001/5d757fe0-502b-4751-8d2e-ce511f0811b0">
This should probably be changed to accept the secondary email, if the primary is not available.

Reply:
Good catch, thank you. Changed it to take the first email, if the primary one is not available.

Reply:
Perfect, now it takes the secondary email, if the primary is not available.
Small thing, but before saving, no email is shown in the frontend:
<img width="998" alt="Before saving 2 0" src="https://github.com/tuwien-csd/damap-backend/assets/80054001/c40095a2-517d-4cf2-b014-1f457e0fff6e">
<img width="989" alt="After saving 2 0" src="https://github.com/tuwien-csd/damap-backend/assets/80054001/fffd188d-1d9e-49ce-a467-bea165e13765">

Reply:
Yes you are right, thanks for looking into it.